core/01_core.js

```diff
@@ -297,11 +349,16 @@
     if (typeof cb != "function") {
       throw new TypeError("expected a function");
     }
-    return op_queue_microtask(() => {
+    const context = AsyncContext.snapshot();
+    op_queue_microtask(() => {
+      let oldContext;
       try {
+        oldContext = AsyncContext.swap(context);
         cb();
       } catch (error) {
         reportExceptionCallback(error);
+      } finally {
+        AsyncContext.swap(oldContext);
```

Comment:
Any chance you could bench this change? Maybe it should be conditional and not use `AsyncContext` if nothing requested it?

Reply:
this is where the upstream v8 changes are going to be important, so benchmarking it now doesn't give us much information.